tooling-docker/src/main/java/org/jboss/eap/qe/ts/common/docker/Docker.java

```diff
@@ -77,7 +82,43 @@ protected void start() throws Exception {
             }
         });
 
-        containerReadyCondition.waitUntilReady(containerReadyTimeout);
+        long startTime = System.currentTimeMillis();
+        while (!isContainerReady()) {
+            if (System.currentTimeMillis() - startTime > containerReadyTimeout) {
+                throw new DockerTimeoutException("Container was not ready in timeout : " + containerReadyTimeout + " ms");
+            }
+            // fail fast mechanism in case of malformed docker command, for example bad arguments, invalid format of port mapping, image version,...
+            if (!dockerRunProcess.isAlive() && dockerRunProcess.exitValue() != 0) {
+                throw new DockerException("Starting of docker container using command: \"" + String.join(" ", cmd)
+                        + "\" failed. Check that provided command is correct.");
+            }
+        }
+    }
+
+    private boolean isContainerReady() throws Exception {
+        CompletableFuture<Boolean> condition = new CompletableFuture().supplyAsync(() -> containerReadyCondition.isReady());
+        try {
+            return condition.get(containerReadyTimeout, TimeUnit.MILLISECONDS);
+        } catch (TimeoutException ex) {
+            // in case condition hangs interrupt it so there are no zombie threads
+            condition.cancel(true);
+
+            throw new ContainerReadyConditionException("Provided ContainerReadyCondition.isReady() method took longer than containerReadyTimeout: "
+                    + containerReadyTimeout + " ms. Check it does not hang and does not take longer then containerReadyTimeout. " +
+                    "It's expected that ContainerReadyCondition.isReady() method is short lived (takes less than 1 second).", ex);
+        }
+    }
+
+    private void checkDockerPresent() throws Exception {
```

Comment:
Shouldn't we check the output too? There might be `alias docker=podman` set up on the system (`podman-docker` package [does this for example when installed](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/building_running_and_managing_containers/index#running-containers-without-docker_building-running-and-managing-containers)) and if we are looking specifically for the Docker, this might be an issue.

Reply:
@honza-kasik thanks, this check works with docker and podman as well. I've tried it on RHEL8.

Reply:
Thanks, that is not my concern. The issue is: Do we want to proceed with Podman?

Reply:
Podman is pushed on RHEL8 and at least for this moment we'll align with this. Current tooling depends on basic usage of `docker` command only so we're ok with having `alias docker=podman`.